### PR TITLE
fix: [SPG-203]: update exposing cluster API doc

### PR DESCRIPTION
**Pull request by RajBaviskar**
## Describe your changes

- Exposing cluster API documentation

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/37426)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/37426)

**Review by rohitk-h — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37426)* status: :shipit: complete! ✓

File: 120-ng-manager/src/main/java/io/harness/ng/gitops/resource/ClusterResource.java [hunk 1]
```diff
@@ -253,8 +249,7 @@ public class ClusterResource {
       responses =
       {
         @io.swagger.v3.oas.annotations.responses.ApiResponse(description = "Returns the list of cluster for a Project")
-      },
-      hidden = true)
+      })
   public ResponseDTO<PageResponse<ClusterResponse>>
   list(@Parameter(description = NGCommonEntityConstants.PAGE_PARAM_MESSAGE) @QueryParam(
```
Comment (abhinav-harness) [inline 120-ng-manager/src/main/java/io/harness/ng/gitops/resource/ClusterResource.java:254]: we have line 307 hidden
review all the api docs too with connector and all other params. 
confirm with rohit if all apis are safe

**Review by rohitk-h — DISMISSED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37426)* status: :shipit: complete! ✓

**Comment by yogesh-chauhan** [comment]
*[`120-ng-manager/src/main/java/io/harness/ng/gitops/resource/ClusterResource.java` line 306 at r2](https://reviewable.io/reviews/harness/harness-core/37426#-NBpNdXa2l5CtH4_3_eQ:-NBpNdXa2l5CtH4_3_eR:b-y5s5mh) ([raw file](https://github.com/harness/harness-core/blob/58f4b7a84918e10b7297e873cc140cc36dbdec64/120-ng-manager/src/main/java/io/harness/ng/gitops/resource/ClusterResource.java#L306)):*
> ```Java
>           {
>             @io.swagger.v3.oas.annotations.responses.ApiResponse(description = "Returns the list of cluster from Gitops Service")
>           })
> ```

This need not be exposed. This API was only created for the U to consume
The original API for listing clusters should be from gitops service

**Review by abhinav-harness — COMMENTED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37426)* status: 0 of 1 approvals obtained.

**Review by abhinav-harness — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37426#-:-NCJB1W7ANbPs8lcmca_:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**